### PR TITLE
HPCC-20924 Spark connector to decouple logic and extend dafilsrv client

### PR DESCRIPTION
**Pull request by jpmcmu**

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

File: DataAccess/src/main/java/org/hpccsystems/spark/HpccFileWriter.java [hunk 1]
```diff
@@ -169,96 +135,79 @@ public long saveToHPCC(RDD<Row> scalaRDD, String clusterName, String fileName) t
     * @throws Exception 
     */
     public long saveToHPCC(SparkContext sc, RDD<Row> scalaRDD, String clusterName, String fileName) throws Exception {
+        return saveToHPCC(sc,scalaRDD,clusterName,fileName,CompressionAlgorithm.NONE,false);
```
Comment (rpastrana) [inline DataAccess/src/main/java/org/hpccsystems/spark/HpccFileWriter.java]: Do spark users need to know the type of compression used?

File: DataAccess/src/main/java/org/hpccsystems/spark/HpccFileWriter.java [hunk 2]
```diff
@@ -169,96 +135,79 @@ public long saveToHPCC(RDD<Row> scalaRDD, String clusterName, String fileName) t
...
-        // Extract filePaths for each partition
-        DFUFilePartWrapper[] fileParts = createResult.getFileParts();
-        final String[] partitionFilePaths = new String[fileParts.length];
-        for (int i = 0; i < fileParts.length; i++) {
-            DFUFileCopyWrapper[] filePartCopies = fileParts[i].getCopies();
-            if (filePartCopies.length == 0) {
-                abortFileCreation();
-                throw new Exception("File creation error: File part: " + (i+1) + " does not have an file copies associated with it. Aborting write.");
+        if (hpccPartitions.length != rdd.getNumPartitions()) {
+            rdd.repartition(hpccPartitions.length);
```
Comment (rpastrana) [inline DataAccess/src/main/java/org/hpccsystems/spark/HpccFileWriter.java]: We might want to log the fact we requested the repartition, we might want to provide timing as well.

File: DataAccess/src/main/java/org/hpccsystems/spark/HpccRDD.java [hunk 3]
```diff
@@ -17,6 +17,9 @@
 
 import java.io.Serializable;
 import java.util.Iterator;
+
+import com.esotericsoftware.minlog.Log;
```
Comment (rpastrana) [inline DataAccess/src/main/java/org/hpccsystems/spark/HpccRDD.java]: was this import inadvertent?

File: DataAccess/src/main/java/org/hpccsystems/spark/HpccFile.java [hunk 4]
```diff
@@ -17,48 +17,29 @@
...
 import org.hpccsystems.ws.client.utils.Connection;
-import org.hpccsystems.ws.client.wrappers.wsdfu.DFUFileAccessInfoWrapper;
 
 /**
  * Access to file content on a collection of one or more HPCC
  * clusters.
  *
  */
-public class HpccFile implements Serializable {
+public class HpccFile extends org.hpccsystems.dfs.client.HPCCFile implements Serializable {
```
[the remark this answers is not in the document]
Comment (jpmcmu) [reply DataAccess/src/main/java/org/hpccsystems/spark/HpccFile.java]: @rpastrana We can't rename it and keep backwards compatibility

File: DataAccess/src/main/java/org/hpccsystems/spark/HpccFileWriter.java (hunk 2 above)
Comment (jpmcmu) [reply DataAccess/src/main/java/org/hpccsystems/spark/HpccFileWriter.java]: The re-partition will show up as a task in Spark job output

**Comment by Michael-Gardner** [comment]
Tested this.  Compiles against the new snapshots of wsclient 7.2.0

**Pull request merged**